Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2394,12 +2394,7 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
     @inlinable
     @differentiable(wrt: self)
     func logSumExp(squeezingAxes axes: Tensor<Int32>) -> Tensor {
-        let rawMax = max(alongAxes: axes)
-        let offset = withoutDerivative(at: rawMax) { rawMax in 
-            Tensor<Scalar>(zerosLike: rawMax).replacing(
-                with: rawMax,
-                where: rawMax.isFinite)
-        }
+        let offset = max(alongAxes: axes)
```

Comment:
This PR also removes the `withoutDerivative(at:)`, which is not part of the finiteness check. I don't know what it's for, but the python tf logic also does it, so I assume that it's important for some reason.

Reply:
For reference, tf python logic is here: https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/ops/math_ops.py#L2517

Reply:
I believe the gradients are equivalent whether or not the `withoutDerivative(at:)` is added. I am also not sure why that was there in the first place. Presumably because `zerosLike` is not differentiable?